tests/cli/workflow/balsamic/test_store_housekeeper.py

```diff
@@ -0,0 +1,226 @@
+        case_id=case_id,
+        samples=["sample_case_wgs_single_tumor",],
+    )
+    # WHEN ensuring analysis_finish file is present
+    mock_analysis_finish(
+        root_dir=balsamic_context["BalsamicAnalysisAPI"].balsamic_api.root_dir, case_id=case_id
+    )
+
+    # Make sure nothing is currently stored in Housekeeper, and no analyses stored in ClinicalDB
+    balsamic_context["BalsamicAnalysisAPI"].housekeeper_api = real_housekeeper_api
+    assert not balsamic_context["BalsamicAnalysisAPI"].store.family(case_id).analyses
+
+    # WHEN running command
+    result = cli_runner.invoke(store_housekeeper, [case_id], obj=balsamic_context)
+
+    # THEN bundle should be successfully added to HK and STATUS
+    assert result.exit_code == EXIT_SUCCESS
+    assert "Analysis successfully stored in Housekeeper" in caplog.text
+    assert "Analysis successfully stored in ClinicalDB" in caplog.text
+    assert balsamic_context["BalsamicAnalysisAPI"].store.family(case_id).analyses
+    assert balsamic_context["BalsamicAnalysisAPI"].housekeeper_api.bundle(case_id)
+
+    # TEARDOWN change real housekeeper back to balsamic fixture
+    balsamic_context["BalsamicAnalysisAPI"].housekeeper_api = balsamic_housekeeper
+
+
+def test_valid_case_already_added(
+    cli_runner, balsamic_context, caplog, real_housekeeper_api, balsamic_housekeeper
+):
+    caplog.set_level(logging.ERROR)
+    # GIVEN case-id
+    case_id = "balsamic_case_tgs_single"
+    # SETUP ensuring case config exists where it should be stored
+    mock_config(
+        root_dir=balsamic_context["BalsamicAnalysisAPI"].balsamic_api.root_dir, case_id=case_id
+    )
+    # SETUP ensuring deliverables exists where it should be stored
+    mock_deliverable(
+        root_dir=balsamic_context["BalsamicAnalysisAPI"].balsamic_api.root_dir,
+        case_id=case_id,
+        samples=["sample_case_tgs_single_tumor",],
+    )
+    # SETUP ensuring analysis_finish file is present
+    mock_analysis_finish(
+        root_dir=balsamic_context["BalsamicAnalysisAPI"].balsamic_api.root_dir, case_id=case_id
+    )
+    # SETUP ensure nothing is currently stored in Housekeeper, and no analyses stored in ClinicalDB
+    balsamic_context["BalsamicAnalysisAPI"].housekeeper_api = real_housekeeper_api
+    assert not balsamic_context["BalsamicAnalysisAPI"].store.family(case_id).analyses
+    # SETUP ensure bundles exist by creating them first
+    cli_runner.invoke(store_housekeeper, [case_id], obj=balsamic_context)
+    # WHEN running command
+    result = cli_runner.invoke(store_housekeeper, [case_id], obj=balsamic_context)
+    # THEN command should NOT execute successfully
+    assert result.exit_code != EXIT_SUCCESS
+    # THEN user should be informed that bundle was already added
+    assert "Bundle already added" in caplog.text
+
+    # TEARDOWN change real housekeeper back to balsamic fixture
+    balsamic_context["BalsamicAnalysisAPI"].housekeeper_api = balsamic_housekeeper
```

Comment:
I would suggest to use fixtures with `scope="function"` to avoid stuff like this what happens if the test fails? Then this line will not get executed

Reply:
Done